### PR TITLE
Data types exercise updates

### PR DESCRIPTION
**Pull request by Genyus**
Modifies and extends string and operator exercises

### CODE REVIEW

File: exercises/java-primitives/README.md [hunk 1]
```diff
@@ -5,173 +5,155 @@
...
-[INFO] Total time:  1.319 s
-[INFO] Finished at: 2023-04-29T16:49:21+01:00
+[INFO] Total time:  1.098 s
+[INFO] Finished at: 2023-10-26T02:34:16+01:00
 [INFO] ------------------------------------------------------------------------
 ```
 
+## :pushpin: Operators
+
+FFor this section, open the `src/main/java/com/cbfacademy/operators/` directory. It contains a [Java program][4] with three methods to implement: `concatenate`, `areEqual` and `format`.
```
Comment (sohbaker) [inline exercises/java-primitives/README.md]: Small typo on For

Comment (sohbaker) [reply exercises/java-primitives/README.md]: Are these the correct method names? For the operators exercise it looks like these names should be different.

File: exercises/java-primitives/src/test/java/com/cbfacademy/strings/AppTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        assertThat(result, is(equalTo(expected)));
+    }
+
+    static Stream<Arguments> concatenate_ReturnsConcatenatedString() {
+        return Stream.of(
+                arguments("Red", "Green", "Blue", "RedGreenBlue"),
+                arguments("one", "two", "three", "onetwothree"),
+                arguments("QUICK", "BROWN", "FOX", "QUICKBROWNFOX"),
+                arguments("www.", "google.", "com", "www.google.com"),
+                arguments("Intro ", "to ", "Java", "Intro to Java"));
```
Comment (sohbaker) [inline exercises/java-primitives/src/test/java/com/cbfacademy/strings/AppTest.java:33]: A thought for a future iteration of this. There could be some value in another exercise which requires using the `String.join()` method, as in real life they're unlikely to receive strings that already have spaces in the appropriate place.

**Review by sohbaker — CHANGES_REQUESTED** [review]
Some changes are needed on the Operators README. Otherwise LGTM.

File: exercises/java-primitives/src/test/java/com/cbfacademy/strings/AppTest.java (hunk 2 above)
Comment (Genyus) [reply exercises/java-primitives/src/test/java/com/cbfacademy/strings/AppTest.java:33]: Makes sense

**Review by sohbaker — APPROVED** [review]

**Pull request merged**